### ISSUE TITLE
TypeError: 'NoneType' object is not iterable

### ISSUE THREAD

**Issue by lifower525** [Toblerity/Shapely#585]
I encountered an error and I don't know how to solve it.

```python traceback
Traceback (most recent call last):
  File "t_test.py", line 8, in <module>
    split(line, multi_point)
  File "/home/lihongyuan02/.pyenv/versions/3.6.5/lib/python3.6/site-packages/shapely/ops.py", line 474, in split
    return GeometryCollection(split_func(geom, splitter))
  File "/home/lihongyuan02/.pyenv/versions/3.6.5/lib/python3.6/site-packages/shapely/ops.py", line 416, in _split_line_with_multipoint
    new_chunks.extend(SplitOp._split_line_with_point(chunk, pt))
TypeError: 'NoneType' object is not iterable
```

## Steps to reproduce the problem.

```python
line = shape(json.loads('{"type": "LineString", "coordinates": [[116.79152, 40.75], [116.78504, 40.75604], [116.78276, 40.75429], [116.78191, 40.75], [116.79152, 40.75]]}'))
multi_point = shape(json.loads('{"type": "MultiPoint", "coordinates": [[116.78524952992561, 40.75584469741501], [116.7829479942355, 40.75]]}'))
split(line, multi_point)
```

## Operating system

(centos 7)

## Shapely version and provenance

(1.6.4.post1 installed from PyPI using pip)

thanks

**Comment by kannes** [comment]
Please put your traceback and code into code blocks for nicer formatting. :)

Reproduced on Archlinux with `Shapely==1.6.4.post1` installed via `pip install --user --no-binary --upgrade shapely`

The first point lies on the line, the other does not.

```
for point in multi_point:
    print(point, line.intersects(point))
```
> POINT (116.7852495299256 40.75584469741501) False
> POINT (116.7829479942355 40.75) True

A made-up, similar case works fine:
```
ls = LineString(((0,0),(1,0)))
p1 = Point(0.1,0)
p2 = Point(0.4,1)  # not on the line
mp = MultiPoint((p1, p2))
s = split(ls, mp)
print(s)
```
> GEOMETRYCOLLECTION (LINESTRING (0 0, 0.1 0), LINESTRING (0.1 0, 1 0))

**Comment by snorfalorpagus** [comment]
The problem here is that the algorithm used to split a line by a point is flawed when the line is a closed ring and the point is on the last segment of the line. Here's a minimal example that triggers the issue.

```python
from shapely.geometry import MultiPoint, LineString
from shapely import ops

line = LineString([[0,0],[0,1],[1,1],[1,0],[0,0]])
point = MultiPoint([[0.5, 0]])

result = ops.split(line, point)

print(result)
```

It shouldn't  be too difficult to rework the existing algorithm to deal with this case. There is already a comment in the function that the distance calculation could be improved.

**Comment by lifower525** [comment]
Thank you for answering my doubts.

**Comment by cjqfeng** [comment]
I have same problem in 1.7a1
when the point is on the last segment 
pd = line.project(Point(p)) always return 0
to solve this when reach the last segment set pd to line.length

**Comment by snorfalorpagus** [comment]
This has been fixed in PR #624 but it hasn't been merged yet and as such isn't in the 1.7a1 release.
